client/state/connectionMiddleware/middleware.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import uniqueId from 'lodash/utility/uniqueId';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isOffline,
+	isOnline
+} from 'state/application/selectors';
+import {
+	OFFLINE_QUEUE_ACTION,
+	CONNECTION_RESTORED,
+	OFFLINE_QUEUE_REMOVE,
+} from 'state/action-types';
+const ACTION_TIMEOUT = 10000;
+
+const connections = {};
+
+let actionTimeout = null;
+
+//Prob should be in actions
+function queueAction( action ) {
+	return {
+		type: OFFLINE_QUEUE_ACTION,
+		id: uniqueId(),
+		dispatchedAt: ( new Date() ).getTime(),
+		action: action
```

Comment:
@gwwar I think if I write something offline, as an example, when I get back online that action should be of higher priority compared to any number of inconsequential fetches. Any action with high priority should be self contained, or if it depended on other actions, those need to be run sequentially. I agree this adds complexity, perhaps we'd like to drop inconsequential actions from the queue more aggressively rather than try to bump important ones.